### ISSUE TITLE
Error Handling

### ISSUE THREAD

**Issue by mhmd-android** [Kaaveh/ComposeNews#205]
Hi, dear @Kaaveh 
I am currently examining and creating an issue related to improving the error-handling mechanism within the system. This issue aims to enhance error handling for better user experience and application robustness.

**Comment by Kaaveh** [comment]
Hey @mhmd-android , awesome! Feel free to make PR!

**Comment by mhmd-android** [comment]
Hi @Kaaveh
I hope this message finds you well. I’ve been considering the integration of a **CustomFactoryAdapter** to enhance our Retrofit request handling. The goal is to sharpen our focus on error handling to improve user experience and bolster the robustness of our application.
However, I must bring to your attention that with our future plans to transition towards **Ktor**, the introduction of this adapter may entail substantial changes. **This could affect both our business logic and testing procedures**. I wanted to lay out this information transparently to deliberate whether the benefits of incorporating the **CustomFactoryAdapter** now would outweigh the potential overhead in the near future.
Your insights on this matter would be invaluable. Should we proceed with this implementation considering the outlined scenario?

Accidentally unassigned myself from this issue yesterday. It was a mistake. Can I be reassigned? Thanks.

**Comment by Kaaveh** [comment]
Hey @mhmd-android , sorry for my late response!
Yah, go ahead! 🙌🏻

**Comment by mhmd-android** [comment]
Hey @Kaaveh

**Comment by mhmd-android** [comment]
Hey @Kaaveh 
No problem, ok, I'll do it, thanks

**Comment by Kaaveh** [comment]
Hey @mhmd-android 
Any update about this task?

**Comment by mhmd-android** [comment]
Hey @Kaaveh
I hope this message finds you well. I wanted to update you on the task I was assigned regarding the error handling for the application. Initially, I planned to complete this using Retrofit and implement a **customFactoryAdapter** to handle the errors. However, I have noticed that **Retrofit has now been replaced with Koin** in the project. Given this change, I will need to consider alternative approaches to implement the error handling effectively. I wanted to inform you about this update and assure you that I will complete it as soon as possible
